commands.py

```diff
@@ -1057,4 +1057,41 @@ def block(bot, event, username, *args):
         from handlers import MessageHandler
 
         MessageHandler.blocked_list.append(u.id_)
-        bot.send_message(event.conv, "Blocked User: {}".format(u.full_name))
\ No newline at end of file
+        bot.send_message(event.conv, "Blocked User: {}".format(u.full_name))
+
+
+@command.register
+def catfacts(bot, event, *args):
+    if ''.join(args) == '?':
+        segments = [hangups.ChatMessageSegment('Catfacts', is_bold=True),
+                    hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK),
+                    hangups.ChatMessageSegment('Usage: /catfacts'),
+                    hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK),
+                    hangups.ChatMessageSegment('Purpose: The more you know.')]
+        bot.send_message_segments(event.conv, segments)
+    else:
+        f = urllib.request.urlopen('http://catfacts-api.appspot.com/api/facts')
+        json_string = f.read()
+        parsed_json = json.loads(json_string.decode())
+        fact = parsed_json["facts"][0]
+        bot.send_message(event.conv, fact)
+
+@command.register
+def insult(bot, event, *args):
+    if ''.join(args) == '?':
+        segments = [hangups.ChatMessageSegment('Insult', is_bold=True),
+                    hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK),
+                    hangups.ChatMessageSegment('Usage: /insult <name>'),
+                    hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK),
+                    hangups.ChatMessageSegment('Purpose: To be mean without being creative.')]
+        bot.send_message_segments(event.conv, segments)
+    else:
+        personal = " ".join(args)+", "
```

Comment:
I like this. For the record, you can get the user's name by using event.user.full_name. You can probably reliably split that by spaces to just get the first name, too. Just a guess, though.

But, as much as I do like it, I don't think this is a good fit for the bot. I'm trying to keep it more functional than funny. That being said, I'll be taking out a lot of the more funny stuff (haiku, MURICA, etc) soon.

Don't let this stop you from submitting more pull requests in the future, though!